core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -34,12 +36,53 @@ abstract class SparkFunSuite
   with Logging {
 // scalastyle:on
 
+  val threadWhiteList = Set(
+    /**
+     * Netty related threads.
```

Comment:
But why are these whitelisted?

Reply:
One example is netty-rpc... related internal threads where we have no control what's going on in the background.

Reply:
Explanation added.